itwm_example/mco/optimizers/optimizers.py

```diff
@@ -285,7 +323,7 @@ def optimize(self):
         f = MultiobjectiveFunction(
             multiobjective_function=self._score, upper_bounds=upper_bounds
         )
-        instrumentation = self._create_instrumentation()
+        instrumentation = self._assemble_instrumentation()
```

Comment:
This is a general comment - from the `MultiobjectiveFunction.compute_aggregate_loss` source code, new data will only be appended to `f._points` if they are accepted by a hypervolume criteria. Therefore out `budget` attribute will not determine the number of points reported but rather the number of calculations attempted, and so it is possible to report no data to the WfManager if the budget is too small.

Could we either make this clearer in the UI or think of a way to hard code the number of expected points to return (`while len(f._points) < self.budget`) ?

Reply:
Thank you, I changed the `budget` name and UI label appropriately. Regarding the `while len(f._points) < self.budget`: as we discussed, this may (and probably almost certainly will) result in an unexpected behaviour. We can't guarantee that that many points exist and can be found.

Reply:
Yes, I agree - changing the UI is more appropriate in this situation